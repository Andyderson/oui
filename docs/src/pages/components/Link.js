
  import React from 'react';
  import ComponentContent from '../../layouts/templates/ComponentContent';

   import data from '../../../../data/components/Link/react.json'; const examples = null; const readme = null; const sassExamples = null;

  let dataObject = {
    react: data ? data : null, 
    examples: examples ? examples : null,
    readme: readme ? readme : null,
    sass: sassExamples ? sassExamples.example : null,
  }
  
  class LinkComponent extends React.Component {
  
    render() {
      return (
        <ComponentContent data={ dataObject } />
      );
    };
  }
  
  export default LinkComponent
