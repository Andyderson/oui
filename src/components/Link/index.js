import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Link = ({
  children,
  href,
  onClick,
  isDisabled = false,
  style,
  testSection,
}) => {
  const classes = classNames({
    ['link']: true,
    [`link--${style}`]: style,
    ['link--disabled']: isDisabled,
    'pointer-events--none': isDisabled,
  });

  return (
    <a
      data-ui-component={ true }
      href={ href }
      className={ classes }
      data-test-section={ testSection }
      data-track-id={ testSection }
      onClick={ onClick }
      disabled={ isDisabled }>
      { children }
    </a>
  );
};

Link.propTypes = {
  /** Text within the button */
  children: PropTypes.node.isRequired,
  /** Describes buttons that have an icon but no text */
  href: PropTypes.string,
  /** Prevent users from interacting with the button */
  isDisabled: PropTypes.bool,
  /** Make the button act as a submit button */
  onClick: PropTypes.func,
  /** Various height and width options */
  style: PropTypes.oneOf([
    'default',
    'dark',
    'muted',
    'bad-news',
    'reverse',
  ]),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
};

export default Link;
