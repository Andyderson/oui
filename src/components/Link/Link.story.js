import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, select } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Link from './index.js';

const stories = storiesOf('Link', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));
stories
  .add('default', withInfo()(() => {
    return (
      <ul>
        <li>
          <Link
            href={ text('href', 'http://google.com') }
            isDisabled={ boolean('isDisabled', false) }
            style={ select('style',
              {
                default: 'default',
                dark: 'dark',
                muted: 'muted',
                'bad-news': 'bad-news',
                reverse: 'reverse',
              }, 'default') }>
            { text('children', 'Link') }
          </Link>
        </li>
      </ul>
    );
  }))
  .add('new window', withInfo()(() => {
    return (
      <ul>
        <li>
          <Link
            href={ text('href', 'http://google.com') }
            newWindow={ boolean('newWindow', true) }
            style={ select('style',
              {
                default: 'default',
                dark: 'dark',
                muted: 'muted',
                'bad-news': 'bad-news',
                reverse: 'reverse',
              }, 'default') }>
            { text('children', 'Link') }
          </Link>
        </li>
      </ul>
    );
  }))
  .add('all links', () => {
    return (
      <ul>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } onClick={ action('click blocklist item home') }>{ text('default', 'Link') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="dark">{ text('dark', 'Link Dark') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="muted">{ text('muted', 'Link Muted') }</Link></li>
        <li><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="bad-news">{ text('bad-news', 'Link Bad News') }</Link></li>
        <li className="background--brand"><Link href="http://google.com" isDisabled={ boolean('isDisabled', false) } style="reverse">{ text('reverse', 'Link Reverse') }</Link></li>
      </ul>
    );
  })
  .add('all disabled', () => {
    return (
      <ul>
        <li><Link href="http://google.com" isDisabled={ true }>Link</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="dark">Link Dark</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="muted">Link Muted</Link></li>
        <li><Link href="http://google.com" isDisabled={ true } style="bad-news">Link Bad News</Link></li>
        <li className="background--brand"><Link href="http://google.com" isDisabled={ true } style="reverse">Link Reverse</Link></li>
      </ul>
    );
  })
  .add('onClick', () => {
    return (
      <ul>
        <li>
          <Link onClick={ action('Clicked a link...') }>
            Click Me!
          </Link>
        </li>
      </ul>
    );
  });
