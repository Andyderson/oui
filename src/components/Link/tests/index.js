import React from 'react';
import Link from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Link', () => {
  it('should render disabled', () => {
    const output = shallow(
      <Link
        href='http://google.com'
        isDisabled={ true }>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render style classes', () => {
    const output = shallow(
      <Link
        style='muted'>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render onclick', () => {
    const output = shallow(
      <Link
        onClick={ function() {} }>
        Test Link
      </Link>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});
