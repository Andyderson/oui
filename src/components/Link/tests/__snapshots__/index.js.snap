// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropdown should render disabled 1`] = `
<a
  className="link link--disabled pointer-events--none"
  data-ui-component={true}
  disabled={true}
  href="http://google.com"
>
  Test Link
</a>
`;

exports[`components/Dropdown should render onclick 1`] = `
<a
  className="link"
  data-ui-component={true}
  onClick={[Function]}
>
  Test Link
</a>
`;

exports[`components/Dropdown should render style classes 1`] = `
<a
  className="link link--muted"
  data-ui-component={true}
>
  Test Link
</a>
`;
